pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy

```diff
@@ -796,16 +796,21 @@ public class Utils {
         Set<String> previousTriggers = new HashSet<>()
         Set<String> previousParameters = new HashSet<>()
 
-        DeclarativeJobPropertyTrackerAction previousAction = null
-        WorkflowRun previousBuild = r.getPreviousBuild()
-        if (previousBuild != null) {
-            previousAction = previousBuild.getAction(DeclarativeJobPropertyTrackerAction.class)
-            if (previousAction != null) {
-                previousProperties.addAll(previousAction.getJobProperties())
-                previousTriggers.addAll(previousAction.getTriggers())
-                previousParameters.addAll(previousAction.getParameters())
+        // First, use the action from the job if it's present.
+        DeclarativeJobPropertyTrackerAction previousAction = j.getAction(DeclarativeJobPropertyTrackerAction.class)
```

Comment:
And even if `pipeline-model-definition` were to reuse an API from `workflow-multibranch` and the changes released simultaneously, there would be some people who would update `workflow-multibranch` first, so we always have the possibility of a two-stage upgrade.

What is the deal with `Trigger`s and `ParameterDefinition`s? My understanding of Declarative was that these were kept in dedicated sections rather than `options` for UX reasons but internally it was all boiling down to the same `properties` call. Am I missing some subtlety?